cmd/root.go

```diff
@@ -150,12 +150,6 @@ func functionWithOverrides(root string, overrides functionOverrides) (f faas.Fun
 		}
 	}
 
-	err =  f.WriteConfig()
-	if err != nil {
-		return
-	}
-
-	f, err = faas.NewFunction(root)
```

Comment:
I thought this was strange when I encountered it initially. I think I like this change, but I think we should call it out explicitly here, that the behavior is changed. When `functionWithOverrides()` is called, the overrides are __not__ persisted. That needs to be handled explicitly in each command if it's the desired behavior.

Reply:
I'll update the function description. The original solution caused that the `faas deploy` command executed outside of any project directory, created the empty `faas.yaml` file in the directory.
This change makes the `functionWithOverrides()` more clean imho.